apmproxy/receiver.go

```diff
@@ -87,9 +93,39 @@ func (c *Client) handleInfoRequest() (func(w http.ResponseWriter, r *http.Reques
 	customTransport.ResponseHeaderTimeout = c.client.Timeout
 	reverseProxy.Transport = customTransport
 
+	reverseProxy.ModifyResponse = func(rsp *http.Response) error {
+		if rsp.StatusCode != http.StatusOK {
```

Comment:
```suggestion
		if rsp.StatusCode == http.StatusOK {
			return nil
		}
```

Nit: return early to improve readability of the following code.

Reply:
The `ModifyResponse` func has been removed so this doesn't apply anymore.